### ISSUE TITLE
clean up iframe message passing code

### ISSUE THREAD

**Issue by bcolloran** [iodide-project/iodide#1450]
per @hamilton's comment here: https://github.com/iodide-project/iodide/pull/1428#discussion_r251545494

One option to investigate: this is a pretty slim lib that seems to handle iframe communication using the style hamilton describes, as well handling and handshakes and stuff -- https://github.com/dollarshaveclub/postmate  Maybe we could build off that?...

**Comment by hamilton** [comment]
@bcolloran do you think at this point we've effectively cleaned up the message passing code?

**Comment by bcolloran** [comment]
not really. it's not on fire ATM, but we still haven't done anything as nice as what you requested in that comment a while back, which i think remains a good thing to shoot for. it's still messy and could use clean up, so we can keep this open to track the idea behind your comment... but i think we share the sentiment that it's not a high priority
